tools/mage/general.go

```diff
@@ -33,3 +34,11 @@ func targetError(err error) error {
 	}
 	return nil
 }
+
+func pathExists(filename string) bool {
```

Comment:
Why not `target.Path` or `target.Dir`?

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/pull/2764#discussion_r442275560

I don't see how that would work. `target.Path` uses modtimes, which cannot be used in the CI caching scenario. It can't work because the checked out files will be modtimed to the current time, meaning that all target checks are returning true, even though a cached value should be used.

Again, if the github action runner runs a target check against a file that has been cached by the action itself, we cannot use the mage target checks, since those would always return `true`, because the modtimes of the cached files will always be older than the freshly checked out source files.

Reply:
https://github.com/magefile/mage/blob/310e198ebd9303cd2c876d96e79de954915f60a7/target/target.go#L27-L30
These checks check existence first. If you care that a path exists only, do not specify dependencies to the call.
This function is redundant.